velox/functions/sparksql/tests/ArrayMaxTest.cpp

```diff
@@ -0,0 +1,272 @@
+         {101, 102, 103, std::numeric_limits<T>::max(), std::nullopt},
+         {std::nullopt, -1, -2, -3, std::numeric_limits<T>::min()},
+         {},
+         {std::nullopt}});
+
+    auto expected = makeNullableFlatVector<T>(
+        {std::numeric_limits<T>::max(),
+         std::numeric_limits<T>::max(),
+         std::numeric_limits<T>::max(),
+         -1,
+         std::nullopt,
+         std::nullopt});
+
+    testArrayMax(input, expected);
+  }
+
+  void testNoNulls() {
+    auto input = makeArrayVector<T>(
+        {{std::numeric_limits<T>::min(), 0, 1, 2, 3, 4},
+         {std::numeric_limits<T>::max(), 3, 2, 1, 0, -1, -2},
+         {std::numeric_limits<T>::max(),
+          101,
+          102,
+          103,
+          104,
+          105,
+          std::numeric_limits<T>::max()},
+         {}});
+
+    auto expected = makeNullableFlatVector<T>(
+        {4,
+         std::numeric_limits<T>::max(),
+         std::numeric_limits<T>::max(),
+         std::nullopt});
+
+    testArrayMax(input, expected);
+  }
+};
+
+template <typename Type>
+class ArrayMaxFloatingPointTest : public FunctionBaseTest {
+ public:
+  using T = typename Type::NativeType::NativeType;
+  static constexpr T kMin = std::numeric_limits<T>::min();
+  static constexpr T kMax = std::numeric_limits<T>::max();
+  static constexpr T kNaN = std::numeric_limits<T>::quiet_NaN();
+
+  void testArrayMax(VectorPtr input, VectorPtr expected) {
+    auto result =
+        evaluate<SimpleVector<T>>("array_max(C0)", makeRowVector({input}));
+    assertEqualVectors(expected, result);
+  }
+
+  void testWithNulls() {
+    auto input = makeNullableArrayVector<T>(
+        {{0.0000, 0.00001},
+         {std::nullopt, 1.1, 1.11, -2.2, -1.0, kMin},
+         {kMin, -0.0001, -0.0002, -0.0003, kMax, kNaN},
+         {},
+         {kMin, 1.1, 1.22222, 1.33, std::nullopt},
+         {-0.00001, -0.0002, 0.0001}});
+
+    auto expected = makeNullableFlatVector<T>(
+        {0.00001, 1.11, kNaN, std::nullopt, 1.33, 0.0001});
+    testArrayMax(input, expected);
+  }
+
+  void testNoNulls() {
+    auto input = makeArrayVector<T>(
+        {{0.0000, 0.00001},
+         {kMax, 1.1, 1.11, -2.2, -1.0, kMin},
+         {kMin, -0.0001, kNaN, -0.0002, -0.0003, kMax},
+         {},
+         {1.1, 1.22222, 1.33},
+         {-0.00001, -0.0002, 0.0001}});
+
+    auto expected = makeNullableFlatVector<T>(
+        {0.00001, kMax, kNaN, std::nullopt, 1.33, 0.0001});
+    testArrayMax(input, expected);
+  }
+};
+
+TYPED_TEST_SUITE(ArrayMaxIntegralTest, FunctionBaseTest::IntegralTypes);
+
+TYPED_TEST(ArrayMaxIntegralTest, arrayMaxNullable) {
+  this->testWithNulls();
+}
+
+TYPED_TEST(ArrayMaxIntegralTest, arrayMax) {
+  this->testNoNulls();
+}
+
+TYPED_TEST_SUITE(
+    ArrayMaxFloatingPointTest,
+    FunctionBaseTest::FloatingPointTypes);
+
+TYPED_TEST(ArrayMaxFloatingPointTest, arrayMaxNullable) {
+  this->testWithNulls();
+}
+
+TYPED_TEST(ArrayMaxFloatingPointTest, arrayMax) {
```

Comment:
Let's make sure to test all supported types. Looks like DATE, decimals and complex types might be missing.

Reply:
add test for date